### PR TITLE
OLXCLONE-41:add signup api with otp verification

### PR DESCRIPTION
**Pull request by KVManeeshReddy**

### CODE REVIEW

File: workspaces/server/utilities/otpUtilities.go [hunk 1]
```diff
@@ -12,3 +15,34 @@ func GenerateOtp() int {
...
+	userInfo := models.VerifyUser{}
+	db.Table(userInfo.TableName()).Where(fmt.Sprintf("%s = ?", constants.MAIL_ID), mailid).Select("otp").Scan(&userInfo)
+	if userInfo.OTP != constants.OTP_DEFAULT {
+		if userInfo.OTP == 0 {
+			return -1
+		}
+		otp = userInfo.OTP
+	} else {
+		otp = GenerateOtp()
+		db.Table(userInfo.TableName()).Create(userInfo)
```
[the remark this answers is not in the document]
Comment (KVManeeshReddy) [reply workspaces/server/utilities/otpUtilities.go:31]: Changed functionality

File: workspaces/server/handlers/userVerificationHandler.go [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+	}
+	mailVerificationObject := createMailVerificationResponse(success)
+	if mailVerificationObject != nil {
+		utilities.WriteJsonResponse(writer, http.StatusOK, mailVerificationObject)
+	}
+}
+
+func UserVerificationHandler(writer http.ResponseWriter, request *http.Request) {
+	var userVerificationReqBodyObject models.VerifyUser
+	userVerificationReqBodyObject.OTP = constants.OTP_DEFAULT
```
[the remark this answers is not in the document]
Comment (KVManeeshReddy) [reply workspaces/server/handlers/userVerificationHandler.go:44]: removed

File: workspaces/server/handlers/forgotPasswordHandler.go [hunk 3]
```diff
@@ -24,31 +24,21 @@ func createforgotPasswordResponse(otpsent bool) []byte {
...
 }
 
 func ForgotPasswordHandler(writer http.ResponseWriter, request *http.Request) {
 	var reqBodyObject forgotPasswordRequestBody
 	utilities.ParseRequestBody(request, &reqBodyObject)
-	success := sendOtp(reqBodyObject)
+	otp := utilities.SendOtp(reqBodyObject.Mailid)
+	success := false
+	if(otp != -1) {
+		updateUserOTP(reqBodyObject.Mailid, otp)
```
[the remark this answers is not in the document]
Comment (KVManeeshReddy) [reply workspaces/server/handlers/forgotPasswordHandler.go:39]: not clear

File: workspaces/server/constants/dbconstants.go [hunk 4]
```diff
@@ -6,7 +6,10 @@ const DB_USER = "dbuname"
...
 const DB_NAME = "dbname"
 
-const Password = "password" //db password column
-const Mailid = "mailid"     //db mailid column
-
 const OTP_DEFAULT = -1
+
+const Password 		= 	"password" //db password column
+const Mailid 		= 	"mailid"     //db mailid column
+const FirstName 	= 	"Maneesh"			//db firstName column
```
[the remark this answers is not in the document]
Comment (KVManeeshReddy) [reply workspaces/server/constants/dbconstants.go:13]: updated constants

**Review by JayachandraGunturu — APPROVED** [review]

**Pull request merged**